src/MessageTrait.php

```diff
@@ -160,6 +151,23 @@ private function setHeaders(array $headers)
         }
     }
 
+    private function normalizeHeaderValue($value)
+    {
+        if (!is_array($value)) {
+            if (!is_string($value)) {
+                throw new \InvalidArgumentException('Header value must be a string or an array of strings.');
```

Comment:
In theory that's correct. But I think it's too risky to add this validation in a minor version, e.g. `withHeader('Api-Version', 1)` would suddenly break for people. Before the integer was simply casted to string which seems like a sensible behavior.
So I would remove this validation for now. It might be something to consider for 2.0.

Reply:
https://github.com/zendframework/zend-diactoros/blob/9a4b4a3c78c1a65e6096fb1f37ab424f33c68821/src/HeaderSecurity.php#L138

Reply:
Looks like this change leads to unexpected exceptions. 
`$request->getBody()->getSize()` returns integer: 
https://github.com/guzzle/guzzle/blob/master/src/PrepareBodyMiddleware.php#L57
and any request leads to throwing that exception.